packages/worker/src/workflow/workflow-worker-thread.ts

```diff
@@ -0,0 +1,91 @@
+import { isMainThread, parentPort as parentPortOrNull } from 'worker_threads';
+import { IllegalStateError } from '@temporalio/common';
+import { VMWorkflow, VMWorkflowCreator } from './vm';
+import { WorkerThreadRequest } from './workflow-worker-thread/input';
+import { WorkerThreadResponse } from './workflow-worker-thread/output';
+import { coresdk } from '@temporalio/proto';
+
+if (isMainThread) {
+  throw new IllegalStateError(`Imported ${__filename} from main thread`);
+}
+
+if (parentPortOrNull === null) {
+  throw new TypeError(`${__filename} got a null parentPort`);
+}
+
+// Create a new parentPort variable that is not nullable to please TS
+const parentPort = parentPortOrNull;
+
+function respond(response: WorkerThreadResponse): void {
+  return parentPort.postMessage(response);
+}
+
+function respondOk(requestId: BigInt): void {
+  return respond({ requestId, result: { type: 'ok' } });
+}
+
+let workflowCreator: VMWorkflowCreator | undefined;
+const workflowByRunId = new Map<string, VMWorkflow>();
+
+parentPort.on('message', async ({ requestId, input }: WorkerThreadRequest) => {
+  try {
+    switch (input.type) {
```

Comment:
I forget, will TypeScript complain here if all types don't have case clauses? Otherwise, might you need a `default` to catch it at runtime?

Reply:
It doesn't complain unless you promise to return something, I'll refactor this so TS covers all cases.